daemon/k8s_watcher.go

```diff
@@ -1383,8 +1316,35 @@ func updateCNPNodeStatus(ciliumNPClient clientset.Interface, cnp *cilium_v2.Cili
 			_, err2 = ciliumNPClient.CiliumV2().CiliumNetworkPolicies(ns).Patch(cnp.GetName(), types.JSONPatchType, createStatusAndNodePatchJSON, "status")
 		}
 	default:
-		cnp.SetPolicyStatus(nodeName, cnpns)
-		_, err2 = ciliumNPClient.CiliumV2().CiliumNetworkPolicies(ns).Update(cnp)
+		// During testing it was noticed kube-apiserver got panics when this
+		// patch was performed, for this reason we will not use node PatchJson.
+		_, err2 = ciliumNPClient.CiliumV2().CiliumNetworkPolicies(ns).Patch(cnp.GetName(), types.JSONPatchType, createStatusAndNodePatchJSON)
```

Comment:
Same here, please do not use `err` and `err2` as names. Please use separate error names for the different errors generated, because it will make the code a lot more readable.